modules/hivemq-edge-module-http/src/main/java/com/hivemq/edge/adapters/http/HttpProtocolAdapter.java

```diff
@@ -150,14 +150,10 @@ public void poll(
 
         final HttpRequest.Builder builder = HttpRequest.newBuilder();
         builder.uri(URI.create(adapterConfig.getUrl()));
-        //-- Ensure we apply a reasonable timeout so we don't hang threads
-        Integer timeout = adapterConfig.getHttpConnectTimeout();
-        timeout = timeout == null ? HttpAdapterConstants.DEFAULT_TIMEOUT_SECONDS : timeout;
-        timeout = Math.max(timeout, HttpAdapterConstants.MAX_TIMEOUT_SECONDS);
-        builder.timeout(Duration.ofSeconds(timeout));
+        builder.timeout(Duration.ofSeconds(adapterConfig.getHttpConnectTimeoutSeconds()));
         builder.setHeader(USER_AGENT_HEADER, String.format("HiveMQ-Edge; %s", version));
 
-        if (adapterConfig.getHttpHeaders() != null && !adapterConfig.getHttpHeaders().isEmpty()) {
+        if (!adapterConfig.getHttpHeaders().isEmpty()) {
```

Comment:
I think we could remove the check. If it is empty the foreach will not be executed.